rfc/rfc-0018-workload-report-artifact-tree.md

```diff
@@ -1,3 +1,202 @@
+        resource:
+          # name of the resource in the supply chain
+          resource-name: <:string>
+          # GVK of the resource
+          kind: <:string>
+          apiVersion: <:string>
+          # name of the resource on the cluster
+          name: <:string>
+          # namespace of the resource
+          namespace: <:string>
+          # resource version of the object
+          resourceVersion: <:string>
+        from:
+          - # id of any artifact(s) that were inputs to the template
+            id: <SHA:string>
+```
+
+example:
+We'll consider a supply chain that stamps out a GitRepository, Runnable (to test the source code), Image, config in a
+configmap, and finally a Kapp App. We'll assume that a new commit was just made, so that the GitRepository has
+just changed url/revision but the Runnable has not yet updated.
+
+Previous Source output:
+uri: https://www.some-site.com/my-project/my-repo
+revision: b974272e27c47a01e7a7da07cf8e4415bdb83dae
+
+Current Source output:
+uri: https://www.some-site.com/my-project/my-repo
+revision: b31d09004503e52e84ff633e547f4d5b40503ab3
+
+```yaml
+status:
+  artifacts:
+    - source:   # <--- new revision output by the GitRepository object
+        id: 146c7d74eb956191487236b579e8e4e68462fc7d97c4f1a4677b0ded39e2a3ca
+        uri: https://www.some-site.com/my-project/my-repo
+        revision: b31d09004503e52e84ff633e547f4d5b40503ab3
+        resource:
+          resource-name: source-provider
+          kind: GitRepository
+          apiVersion: source.toolkit.fluxcd.io/v1beta1
+          name: my-app
+          namespace: my-namespace
+          resourceVersion: "11125094"
+    - source:   # <--- old revision from GitRepository. This source is still reported as another artifact still relies on it
+        id: 23156c7ac2170fe95f85a1ad42522c408e67038f8393eea6cd8551e07457c5d7 # <--- the sha256sum of the passed, revision and uri fields
+        uri: https://www.some-site.com/my-project/my-repo
+        revision: b974272e27c47a01e7a7da07cf8e4415bdb83dae
+        resource:
+          resource-name: source-provider
+          kind: GitRepository
+          apiVersion: source.toolkit.fluxcd.io/v1beta1
+          name: my-app
+          namespace: my-namespace
+          resourceVersion: "11125090"
+    - source:   # <--- source artifact from the Runnable
+        id: 74cb6607d64da1e0324196517340ac7a668042f0e9dfbdd58f8a00a5d0ee9580
+        uri: https://www.some-site.com/my-project/my-repo
+        revision: b974272e27c47a01e7a7da07cf8e4415bdb83dae
+        resource:
+          resource-name: source-tester
+          kind: Runnable
+          apiVersion: carto.run/v1alpha1
+          name: my-app
+          namespace: my-namespace
+          resourceVersion: "11125545"
+        from:
+          - id: 23156c7ac2170fe95f85a1ad42522c408e67038f8393eea6cd8551e07457c5d7 # <--- The id noted above, as the source-tester consumes the source artifact from source-provider
```

Comment:
If a component like `kpack` can generate a new build based on external stimulus (i.e. addition of a new base image), what happens to `from` here?